src/refactorings/change-signature/change-signature.ts

```diff
@@ -114,26 +115,47 @@ function updateCode(
         const args = node.arguments.slice();
         if (args.length) {
           newPositions.forEach((order) => {
-            const arg = node.arguments[order.value.startAt];
-            args[order.value.endAt] = arg;
+            if (order.value.startAt === -1) {
+              // Convert to a valid code.
+              // Without that will trigger invalid "Missing semicolon (n, n)"
+              // That error occurs only for literal objects like: {id: 1, ...}
```

Comment:
Interesting. Thanks for adding the comment. 

Yeah, I understand why it would fail if you just try to parse the value without assigning it to a variable OR passing it as a param to something. That is a reasonable workaround for me.

TBH, the moment I see passing tests and I can't think of more scenarios, I'm happy 😉